Sources/WebSocket/Client.swift

```diff
@@ -0,0 +1,86 @@
+import Dispatch
+import Core
+import Crypto
+import Foundation
+import TCP
+import HTTP
+
+extension WebSocket {
+    /// Create a new WebSocket client in a future.
+    ///
+    /// The future will be completed with the WebSocket connection once the handshake using HTTP is complete.
+    ///
+    /// - parameter hostname: The server's hostname to connect to
+    /// - parameter port: The port to connect to, for HTTP
+    /// - parameter uri: The URI is not officially part of the spec, but could route to a different API on the server
+    /// - parameter queue: The queue on which this websocket will read and write
+    public static func connect(to hostname: String, atPort port: UInt16 = 80, uri: URI = URI(path: "/"), queue: DispatchQueue) throws -> Future<WebSocket> {
```

Comment:
this should match the TCP.Socket connect method parameter naming.

given that `String` is loosely typed, i think `hostname: "foo"` is more clear.